docs/HTTP_SPEC.md

```diff
@@ -14,107 +14,287 @@
+
+## Request Query Parameters
+
+### `filename` (request query parameter)
+
+Same as [Path Gateway](https://specs.ipfs.tech/http-gateways/path-gateway/#filename-request-query-parameter), but only supports `.car` extensions.
+
+_OPTIONAL_. Used to override the downloaded filename. If provided, it will be included in the `Content-Disposition` header.
+
+If provided, the filename extension is _REQUIRED_ and _MUST_ be `.car`. A missing extension, or any other unspecified extension, will respond with a 400 status code.
+
+Example:
+- `filename=my-file.car` will download a CAR with the name filename `my-file.car`
+
+### `format` (request query parameter)
+
+Same as [Path Gateway](https://specs.ipfs.tech/http-gateways/path-gateway/#format-request-query-parameter), but only supports the `car` format.
+
+_OPTIONAL_. Used to specify the response content type.
+
+This is a URL-friendly alternative to providing an [`Accept`](#accept-request-header) header. _OPTIONAL_ only if an `Accept` header value is provided, otherwise this parameter is _REQUIRED_. Unlike the [`Accept`](#accept-request-header), there is no way to specify media type parameter values. Default media type parameter values will be applied.
+
+If provided, the value _MUST_ be `car`. Any other value will respond with a 400 status code.
+
+Example:
+- `format=car` &rarr; `Accept: application/vnd.ipld.car`
+
+### `dag-scope` (request query parameter)
+
+Specified in [IPIP-402](https://github.com/ipfs/specs/pull/402).
+
+_OPTIONAL_. `dag-scope=(block|entity|all)`. Defaults to `all`.
+
+Describes the shape of the DAG fetched at the terminus of the specified path whose blocks are included in the returned CAR file after the blocks required to traverse path segments. Unspecified values will respond with a 400 status code.
+
+- `block`: Only the root block at the end of the path is returned after blocks required to verify the specified path segments.
+
+- `entity`: Returns only the content at the termination of the `{cid}[/path]` specifier, as well as all blocks from the `cid` to the `path` terminus where a `path` is provided. If the content is found to be UnixFS data, the entire UnixFS entity will be included. i.e. if `{cid}[/path]` terminates at a sharded UnixFS file, or a sharded UnixFS directory, the blocks required to reconsititute the entire file, or directory will be included. If the termination is a UnixFS sharded directory, only the full directory will be included, not the full DAG of the directory's contents.
+
+- `all`: Transmit the entire contiguous DAG that begins at the end of the path query, after blocks required to verify path segments.
+
+When present, the `Etag` response header must include a unique prefix based on the passed scope type.
+
+Examples:
+- `dag-scope=block` 
+- `dag-scope=entity`
+- `dag-scope=all`
+
+### `protocols` (request query parameter)
+
+_OPTIONAL_. `protocols=[...bitswap|graphsync|http]`. Defaults to any valid protocol returned by IPNI.
```

Comment:
```suggestion
_OPTIONAL_. `protocols=[...bitswap|graphsync|http]`. Defaults to any valid protocol returned by [IPNI](https://github.com/ipni/specs/blob/main/IPNI.md).
```

Reply:
you may want to look at consistency for the one-of-many options - `protocols=[.|.|.]` is different from `dupes=(y|n)`. `[]` I think is more standard way, `()` is a bit regexy.

Reply:
also not sure about how to represent this as `,`, maybe that's what you're trying to do with `...`? could be something more posixy like:

> `protocols=a[,b,[c...]]]` Where protocols `a`, `b`, `c` are any combination of `bitswap`, `graphsync` and `http`.

but that gets kind of messy!

Reply:
I saw that `()` was used in the ipfs specs so I just continued using it. I wanted to use `[]` to signify a list, but wasn't sure how to show that it should be comma delimited.

Actually now that I'm thinking about this, I think the following is a lot clearer:
```
protocols=[bitswap,graphsync,http]
```

Reply:
Went back and double checked, the only example in the Path Gateway where an example is shown, it's using `<>`. I found the example in one of the IPIPs. I'd also be down for using `<>`, but I like the `[]`.